Integrations/MicrosoftGraphCalendar/MicrosoftGraphCalendar.py

```diff
@@ -0,0 +1,717 @@
+        event_data = event_data[0]
+
+    # add a field that indicates that the event was deleted
+    event_data['Deleted'] = True  # add a field with the members to the event
+    entry_context = {f'{INTEGRATION_CONTEXT_NAME}.Event(val.ID === obj.ID)': event_data}
+
+    human_readable = f'Event was deleted successfully.'
+    return human_readable, entry_context, NO_OUTPUTS
+
+
+def list_calendars_command(client: Client, args: Dict) -> Tuple[str, Dict, Dict]:
+    """
+    Get all the user's calendars (/calendars navigation property)
+
+    Args:
+        client: Client object with request
+        args: Usually demisto.args()
+    """
+    calendar = client.list_calendars(**args)
+
+    calendar_readable, calendar_outputs = parse_calendar(calendar.get('value'))  # type: ignore
+
+    entry_context = {f'{INTEGRATION_CONTEXT_NAME}.Calendar(val.ID === obj.ID)': calendar_outputs}
+    title = 'Calendar:'
+
+    human_readable = tableToMarkdown(
+        name=title,
+        t=calendar_readable,
+        headers=CALENDAR_HEADERS,
+        removeNull=True
+    )
+
+    return human_readable, entry_context, calendar
+
+
+def get_calendar_command(client: Client, args: Dict) -> Tuple[str, Dict, Dict]:
+    """
+    Get the properties and relationships of a calendar object.
+    The calendar can be one for a user, or the default calendar of an Office 365 group.
+
+    Args:
+        client: Client object with request
+        args: Usually demisto.args()
+    """
+    calendar = client.get_calendar(**args)
+
+    calendar_readable, calendar_outputs = parse_calendar(calendar)
+
+    entry_context = {f'{INTEGRATION_CONTEXT_NAME}.Calendar(val.ID === obj.ID)': calendar_outputs}
+    title = 'Calendar:'
+
+    human_readable = tableToMarkdown(
+        name=title,
+        t=calendar_readable,
+        headers=CALENDAR_HEADERS,
+        removeNull=True
+    )
+
+    return human_readable, entry_context, calendar
+
+
+def module_test_function_command(client: Client, args: Dict) -> Tuple[str, Dict, Dict]:
+    """
+    Performs a basic GET request to check if the API is reachable and authentication is successful.
+
+    Args:
+        client: Client object with request
+    """
+    return client.test_function()
+
+
+def main():
+    url = demisto.params().get('url').rstrip('/') + '/v1.0/'
+    tenant = demisto.params().get('tenant_id')
+    auth_and_token_url = demisto.params().get('auth_id').split('@')
+    auth_id = auth_and_token_url[0]
+    enc_key = demisto.params().get('enc_key')
+    verify = not demisto.params().get('insecure', False)
+    proxy = demisto.params().get('proxy', False)
+    default_user = demisto.params().get('default_user')
+
+    if len(auth_and_token_url) != 2:
+        # token_retrieval_url = 'https://oproxy.demisto.ninja/obtain-token'  # disable-secrets-detection
+        token_retrieval = \
+            'https://us-central1-oproxy-dev.cloudfunctions.net/calendar_graph_ProvideAccessTokenFunction'  # todo: prod
```

Comment:
fix and clean this when moving to prod

Reply:
@orenzohar are we ready to move to prod? this is the last thing waiting for approval.